app/views/hostgroups/_form.html.erb

```diff
@@ -27,10 +27,19 @@
       <%= text_f f, :name %>
       <%= textarea_f f, :description, :rows => :auto %>
       <%= hostgroup_puppet_environment_field f %>
-      <%= select_f(f, :compute_profile_id, ComputeProfile.visibles, :id, :name,
+
+      <%= select_f f, :compute_resource_id, accessible_resource(@hostgroup, :compute_resource), :id, :to_label,
+          { :include_blank => blank_or_inherit_f(f, :compute_resource, :inherited_blank_value => _('Bare Metal'), :blank_value => _('Bare Metal')) },
+          { :label => _('Deploy on'),
+            :onchange => 'tfm.hostgroups.computeResourceChanged(this);',
+            :help_inline => :indicator } if SETTINGS[:unattended] %>
+
+      <div id="compute_profile" <%= display?(!@hostgroup.compute_resource_id) %> >
```

Comment:
I hide this to keep consistent behaviour with hosts form. If users just set the compute profile, then they have 'Bare Metal' as default compute resource and that combination does not make much sense to me.